test/clients/http_response_test.rb

```diff
@@ -45,6 +45,15 @@ def test_next_and_prev
         assert_equal("page-info", response.prev_page_info)
         assert_equal("other-page-info", response.next_page_info)
       end
+
+      def test_object_response
```

Comment:
Should we duplicate all the tests in this file with the context_object set to true, to test we are not breaking the original functionality?